### PR TITLE
Nft api search contract metadata

### PR DESCRIPTION
**Pull request by xeno097**
### Changelog

- Adds the `searchContractMetadata` method in the `nft` namespace.
- Refactors unit tests for the `getNftContractMetadata` method.
- Moves the `verifyNftContractMetadata` util function to the `test/test-util.ts` file to reuse it for the `searchContractMetadata` unit tests.
- Adds unit and integration tests for `searchContractMetadata` method.

### CODE REVIEW

File: test/test-util.ts [hunk 1]
```diff
@@ -203,6 +204,23 @@ export function createRawNftContractBaseNft(
   };
 }
 
+export function verifyNftContractMetadata(
+  actualNftContract: NftContract,
+  expectedNftContract: NftContract,
+  address: string,
```
Comment (thebrianchen) [inline test/test-util.ts:210]: why do we need all the additional variables in the params here?

i'm thinking that we would just do
`expect(actualNftContract.address).toEqual(expectedNftContract.address);`

**Review by thebrianchen — CHANGES_REQUESTED** [review]
thanks for submitting this! just a small change and we should be good to go!

File: test/test-util.ts (hunk 1 above)
Comment (xeno097) [reply test/test-util.ts:210]: Playing with the API I saw that every element in the `searchContractMetadata` endpoint response has the same structure as the response from the `getNftContractMetadata` endpoint which was unit tested using the following function 

image
image

So to keep the testing method consistent I decided to reuse the `verifyNftContractMetadata` function for the `searchContractMetadata` tests too.

Comment (thebrianchen) [reply test/test-util.ts:210]: ah I see what you mean, thanks for clarifying!

please leave this as is, i'll refactor this later!

Comment (xeno097) [reply test/test-util.ts:210]: Ok, nice! Thank you!

**Review by thebrianchen — APPROVED** [review]
thanks again!

**Pull request merged**